### ISSUE TITLE
Incremental search does not work

### ISSUE THREAD

**Issue by anissen** [godotengine/godot#29442]
**Godot version:**
Latest master built from source. SHA: 42569f566

**OS/device including version:**
macOS Mojave (10.14), MacBook Air from mid 2012

**Issue description:**
Incremental search does not work. However, it has never worked on my machine so it's not a regression. I've tried in the file tree, the scene tree and the tree for adding new nodes. Pressing any key (a-zA-Z) does nothing.

Expected: That the first item starting with the corresponding character was selected when pressing a key.

**Steps to reproduce:**
Open the project manager and click "Scan". Focus the file dialog tree. Press the key corresponding to the first character of a file or directory shown.

**Minimal reproduction project:**
N/A

--- 

**Observations:**
If I change the existing code (https://github.com/godotengine/godot/blob/master/scene/gui/tree.cpp#L2385-L2394) to the following code, I get functional incremental search (although it seems bit quirky but I guess that's how it's implemented).

```c++
if (k->get_unicode() > 0) {

    _do_incr_search(String::chr(k->get_unicode()));
    accept_event();

    return;
} else if (k->get_scancode() > 0) { // this part is new
    
    _do_incr_search(String::chr(k->get_scancode()));
    accept_event();

    return;
} else {
    if (k->get_scancode() != KEY_SHIFT)
        last_keypress = 0;
}
```

This is probably not the right fix (I also get a crash after doing incremental searches in a few different directories) but may give some clue. When I press 'p', `unicode` is 0 in `k` while `scancode` is 80.

**Comment by Calinou** [comment]
I reported the crash in https://github.com/godotengine/godot/issues/29441.

**Comment by bruvzg** [comment]
Related to #28224, `get_unicode()` always return zero unless IME input node is activated.

**Comment by anissen** [comment]
My issue is fixed by #29464.

Thanks @bruvzg! 🥇
